test_integration/03_redeem_and_unstake/01_redeem_and_unstake.js

```diff
@@ -0,0 +1,269 @@
+            redeemRequest.redeemer,
+            redeemRequest.nonce,
+            redeemRequest.beneficiary,
+            redeemRequest.amount,
+            redeemRequest.gasPrice,
+            redeemRequest.gasLimit,
+            redeemRequest.blockHeight,
+            redeemRequest.hashLock,
+            outboxProof.storageProof[0].serializedProof,
+            { from: originAccounts[0] },
+        );
+
+        event = EventDecoder.getEvents(tx, gateway);
+        // Assert event.
+        ConfirmRedeemIntentAssertion.verify(event, redeemRequest);
+    });
+
+    it('progresses redeem', async () => {
+        const initialBalanceBeforeProgress = await progressRedeemAssertion.captureBalances(
+            redeemRequest.redeemer,
+        );
+
+        const response = await cogateway.progressRedeem(
+            redeemRequest.messageHash,
+            redeemRequest.unlockSecret,
+            { from: redeemRequest.redeemer },
+        );
+
+        const gasConsumptionInProgress = await getTransactionGasConsumption(
+            response,
+            auxiliaryWeb3,
+        );
+        const event = EventDecoder.getEvents(response, cogateway);
+
+        await progressRedeemAssertion.verify(
+            event,
+            redeemRequest,
+            gasConsumptionInProgress,
+            initialBalanceBeforeProgress,
+        );
+    });
+
+    it('progresses unstake', async () => {
+
+        const facilitator = originAccounts[0];
+        const initialBalancesBeforeProgress = await progressUnstakeAssertion.captureBalances(
+            redeemRequest.beneficiary,
+            facilitator,
+        );
+
+        const tx = await gateway.progressUnstake(
+            redeemRequest.messageHash,
+            redeemRequest.unlockSecret,
+            { from: facilitator },
+        );
+
+        const event = EventDecoder.getEvents(tx, gateway);
+
+        await progressUnstakeAssertion.verify(
+            event,
+            redeemRequest,
+            initialBalancesBeforeProgress,
+            facilitator,
+        );
+    });
+});
+
+/**
+ * This approve cogateway for redeem amount by unwraping base token.
+ * @param {Object} ostPrime OSTPrime contract instance.
+ * @param {Object} redeemRequest Redeem request object.
+ * @param {Object }cogateway CoGateway contract instance.
+ * @return {Promise<void>}
+ */
+async function approveCogatewayForRedeemAmount(ostPrime, redeemRequest, cogateway) {
+    await ostPrime.wrap(
+        {
+            value: redeemRequest.amount,
+            from: redeemRequest.redeemer,
+        },
+    );
+
+    await ostPrime.approve(
+        cogateway.address,
+        redeemRequest.amount,
+        { from: redeemRequest.redeemer },
+    );
+}
+
+/**
+ * This returns gas consumption in a transaction.
+ * @param {Object} response Transaction object by truffle.
+ * @param {Web3 } web3
+ * @return {Promise<BN>} gas consumption
+ */
+async function getTransactionGasConsumption(response, web3) {
```

Comment:
As it returns consumed gas times gas price, the name is a little misleading.

Reply:
Updated to `getTransactionFee`
2bb0923